Build is broken on macOS

#### Problem
Seeing these build errors while building on macOS

```
../src/lib/support/CHIPMem.h:67:37: error: redefinition of default argument
extern CHIP_ERROR MemoryInit(void * buf = nullptr, size_t bufSize = 0);
                                    ^     ~~~~~~~
gen/include/support/CHIPMem.h:67:37: note: previous definition is here
extern CHIP_ERROR MemoryInit(void * buf = nullptr, size_t bufSize = 0);
                                    ^     ~~~~~~~
In file included from ../src/lib/support/CHIPArgParser.cpp:49:
In file included from gen/include/support/CHIPMemString.h:29:
In file included from ../src/lib/support/ScopedBuffer.h:27:
../src/lib/support/CHIPMem.h:67:59: error: redefinition of default argument
extern CHIP_ERROR MemoryInit(void * buf = nullptr, size_t bufSize = 0);
                                                          ^         ~
gen/include/support/CHIPMem.h:67:59: note: previous definition is here
extern CHIP_ERROR MemoryInit(void * buf = nullptr, size_t bufSize = 0);
                                                          ^         ~
In file included from ../src/lib/support/CHIPArgParser.cpp:49:
In file included from gen/include/support/CHIPMemString.h:29:
In file included from ../src/lib/support/ScopedBuffer.h:27:
../src/lib/support/CHIPMem.h:166:12: error: redefinition of 'New'
inline T * New(Args &&... args)
           ^
gen/include/support/CHIPMem.h:166:12: note: previous definition is here
inline T * New(Args &&... args)
           ^
In file included from ../src/lib/support/CHIPArgParser.cpp:49:
In file included from gen/include/support/CHIPMemString.h:29:
In file included from ../src/lib/support/ScopedBuffer.h:27:
../src/lib/support/CHIPMem.h:179:13: error: redefinition of 'Delete'
inline void Delete(T * p)
            ^
gen/include/support/CHIPMem.h:179:13: note: previous definition is here
inline void Delete(T * p)
```

 #### Proposed Solution
Fix the build issues.

Seeing this on a clean workspace:

```
In file included from ../src/platform/Darwin/PlatformManagerImpl.cpp:28:
In file included from ../src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp:33:
../src/platform/Linux/MdnsImpl.h:29:10: fatal error: 'avahi-client/client.h' file not found
#include <avahi-client/client.h>
```

Not sure if the errors in the original issue comment are still occurring.

While trying to repro this, I see the same error...

```
In file included from ../src/platform/Darwin/PlatformManagerImpl.cpp:28:
In file included from ../src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp:33:
../src/platform/Linux/MdnsImpl.h:29:10: fatal error: 'avahi-client/client.h' file not found
#include <avahi-client/client.h>
```

I think. both of these are valid errors.

--- a/src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp
+++ b/src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp
@@ -30,7 +30,9 @@
 
 // Include the non-inline definitions for the GenericPlatformManagerImpl<> template,
 // from which the GenericPlatformManagerImpl_POSIX<> template inherits.
+#if CHIP_ENABLE_MDNS
 #include <platform/Linux/MdnsImpl.h>
+#endif
 #include <platform/internal/GenericPlatformManagerImpl.cpp>
 
 #include <system/SystemLayer.h>

